analysis_driver/dataset_scanner.py

```diff
@@ -129,33 +129,15 @@ def _is_valid_dataset(self, dataset):
 
 
 class SampleScanner(DatasetScanner):
-    endpoint = 'aggregate/samples'
+    endpoint = 'samples'
     item_id = 'sample_id'
 
-    def get_dataset(self, name, most_recent_proc=None, data_threshold=None):
-        return SampleDataset(name, most_recent_proc, data_threshold)
-
-    def _get_datasets_for_statuses(self, statuses):
-        self.debug('Creating Datasets for status %s', ', '.join(statuses))
-        datasets = {}
-        for r in self._get_dataset_records_for_statuses(statuses):
-            datasets[r[self.item_id]] = {'record': r}
-
-        if datasets:
-            self.debug('Querying Lims for %s samples', len(datasets))
-            for sample in get_list_of_samples(list(datasets)):
-                req_yield = sample.udf.get('Yield for Quoted Coverage (Gb)') * 1000000000
-                datasets[sanitize_user_id(sample.name)]['threshold'] = req_yield
-            self.debug('Lims query complete')
-
-        return [
-            self.get_dataset(k, v['record'].get('most_recent_proc'), v.get('threshold'))
-            for k, v in datasets.items()
-        ]
+    def get_dataset(self, name, most_recent_proc=None):
+        return SampleDataset(name, most_recent_proc)
```

Comment:
This removes previous optimisation, so a new query will be made for every sample scanned. We should re-add the optimisation later

Reply:
That is correct. However because we're not using aggregate endpoint anymore it is still a faster.
I agree that the optimisation should be added  back in next version and probably extended to run and projects.